### PR TITLE
[Port] Makes gravity event based, QOL additions as well as bugfixes regarding gravity

### PR DESCRIPTION
**Pull request by XeonMations**
## About The Pull Request

Ports:
- [x] https://github.com/tgstation/tgstation/pull/61054
- [x] https://github.com/tgstation/tgstation/pull/74084
- [x] https://github.com/tgstation/tgstation/pull/79785
- [x] https://github.com/tgstation/tgstation/pull/76824
- [x] https://github.com/tgstation/tgstation/pull/74671
- [x] https://github.com/tgstation/tgstation/pull/38043 (I found parts of the code from this PR already in, and some parts missing for some reason)
- [x] https://github.com/tgstation/tgstation/pull/66037 (Gravity related code in living.dm)
- [x] https://github.com/tgstation/tgstation/pull/63791 (Gravity changes)
- [x] https://github.com/tgstation/tgstation/issues/66657 (Even more gravity changes)
- [x] https://github.com/tgstation/tgstation/pull/75333/
- [x] https://github.com/tgstation/tgstation/pull/76764/
- [x] https://github.com/tgstation/tgstation/pull/71664
- [x] https://github.com/tgstation/tgstation/pull/74285 (Partially)

## Why It's Good For The Game

Bugfixes are good, as well as fixing there being constant checks for people's gravity. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/user-attachments/assets/de864f1a-1ce6-4b87-8c15-3329e4350468

https://github.com/user-attachments/assets/ccb17cd0-8c85-480e-8139-c0c074b4c475

https://github.com/user-attachments/assets/2e88f2ed-a51e-463b-b07e-588f5e7e0088

</details>

## Changelog
:cl: Ghommie, LemonInTheDark, Kylerace, vinylspiders, MrMelbert, Fikou, AnturK, ShizCalev, OrionTheFox, XeonMations
fix: Gravity generator now shows repair instructions on examine
add: Proper functionality for upside down gravity, and high gravity situations
refactor: Gravity is now event based, meaning you should have less downtime from moving into a weightless area and your gravity status changing.
/:cl:

### CODE REVIEW

File: code/datums/proximity_monitor/fields/gravity.dm [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 /datum/proximity_monitor/advanced/gravity/setup_field_turf(turf/T)
 	. = ..()
-	T.AddElement(/datum/element/forced_gravity, gravity_value)
+	T.AddElement(/datum/element/forced_gravity, gravity_value,, can_override = TRUE)
```
Comment (PowerfulBacon) [inline code/datums/proximity_monitor/fields/gravity.dm]: Is this meant to have 2 commas?

Comment (XeonMations) [reply code/datums/proximity_monitor/fields/gravity.dm]: Oops.

**Review by PowerfulBacon — APPROVED** [review]
Will do some testing on local and merge if this seems okay

**Review by PowerfulBacon — APPROVED** [review]
Seems okay on local testing

**Pull request merged**